crates/biome_fs/src/dir.rs

```diff
@@ -0,0 +1,15 @@
+use directories::ProjectDirs;
+use std::{env, fs, path::PathBuf};
+
+pub fn ensure_cache_dir() -> PathBuf {
+    let project_data_dir = if let Some(proj_dirs) = ProjectDirs::from("dev", "biomejs", "biome") {
+        // Linux: /home/alice/.cache/biome
+        // Win: C:\Users\Alice\AppData\Local\biomejs\biome\cache
+        // Mac: /Users/Alice/Library/Caches/dev.biomejs.biome
+        proj_dirs.cache_dir().to_path_buf()
+    } else {
+        env::temp_dir()
+    };
+    fs::create_dir_all(&project_data_dir).expect("failed to create biome cache directory");
```

Comment:
The temporary directory is already created by the OS, this means that this instruction would fail in case the program uses the temporary directory.

Also, this could fail the second time we start the daemon, because the directories were already created. We should check their existence before the creation, right?

Reply:
According to the [doc](https://doc.rust-lang.org/std/fs/fn.create_dir_all.html), it create a directory and all of its parent components if they are missing.

I just verified with `fs::create_dir_all(&env::temp_dir()).expect("should not fail");` and it does not fail.

Reply:
Thank you. Can we handle the error differently and fallback to use the temporary directory instead? Since we know that it's always present, we should have possible permission issues

Reply:
Sure, fixed.

Reply:
Sorry I wasn't clear enough. I meant **to not use `expect`**, but instead handle `Err` and use the temporary directory

Reply:
Fixed.